client.go

```diff
@@ -123,10 +136,32 @@ func NewClient() *Client {
 	}
 }
 
+// ErrStopRetry is returned by a CheckRetry callback to stop retries, but
+// still have the client return the original error.
+var ErrStopRetry = errors.New("stop retry")
```

Comment:
Instead of having a magic error, I'd rather have the `CheckRetry` return a `bool, error` tuple, the bool indicating if we should stop or not.